R/generateMappingfile.R

```diff
@@ -36,13 +36,18 @@
 #' )
 #' }
 #' @export
-generateMappingfile <- function(templates, targetVar, outputDirectory = "output",
-                                fileName = NULL, remindVar = "r30m44",
-                                remindUnit = "r30m44_unit", targetUnit = "Unit",
-                                factorCol = "r30m44_factor", weightCol = NULL,
-                                spatialCol = "spatial", model = "REMIND-MAgPIE",
-                                commentFileName = NULL) {
-  if (!dir.exists(outputDirectory)) {
+generateMappingfile <- function(templates = NULL, outputDirectory = "output", fileName = NULL, commentFileName = NULL,
+                                targetVar = "Variable", targetUnit = "Unit",
+                                remindVar = "r30m44", remindUnit = "r30m44_unit",
+                                factorCol = "r30m44_factor", weightCol = NULL, spatialCol = "spatial",
+                                model = "REMIND-MAgPIE"
+                                ) {
+  templateNames <- getTemplateNames()
+  templates[templates %in% names(templateNames)] <- templateNames[templates[templates %in% names(templateNames)]]
```

Comment:
line 46 allows to specify `templates` as `c("SHAPE", "NAVIGATE")` and it finds the correct path…